bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/CameraAddress.java

```diff
@@ -25,46 +23,39 @@
 @NonNullByDefault
 public class CameraAddress {
 
-    private final String vpnURL;
-    private final String localURL;
+    private final String vpnURL; // https://prodvpn-eu-2.netatmo.net/restricted/10.255.39.58/f48daf5f0e366bc8f9c397fbc73220e7/MTYyNDQ1NjgwMDrf1ChChrvaKZolVvWtHuuxT8_EiA,,
+    private final boolean local;
+    private @Nullable String localURL; // http://192.168.0.65/f48daf5f0e366bc8f9c397fbc73220e7
 
-    CameraAddress(final String vpnURL, final String localURL) {
+    CameraAddress(String vpnURL, boolean isLocal, @Nullable String localURL) {
         this.vpnURL = vpnURL;
+        this.local = isLocal;
         this.localURL = localURL;
     }
 
     public String getVpnURL() {
         return vpnURL;
     }
 
-    public String getLocalURL() {
+    public @Nullable String getLocalURL() {
         return localURL;
     }
 
+    public boolean isLocal() {
+        return local;
+    }
+
+    public String getStreamUrl(@Nullable String videoId) {
+        return String.format("%s/vod/%s/%s.m3u8", vpnURL, videoId, local ? "index_local" : "index");
```

Comment:
You should consider the case `videoId` is null

Reply:
Corrected appropriately. videoId can not be null here.